src/services/api/routes/communities.ts

```diff
@@ -253,31 +247,54 @@ router.post('/', multer().none(), async function (request: express.Request, resp
 
 	const parentCommunity: HydratedCommunityDocument | null = await getCommunityByTitleID(request.paramPack.title_id);
 	if (!parentCommunity) {
-		respondCommunityNotFound(response);
-		return;
+		return respondCommunityNotFound(response);
 	}
 
 	// TODO - Better error codes, maybe do defaults?
 	const bodyCheck: z.SafeParseReturnType<CreateNewCommunityBody, CreateNewCommunityBody> = createNewCommunitySchema.safeParse(request.body);
 	if (!bodyCheck.success) {
-		response.send(xmlbuilder.create({
-			result: {
-				has_error: 1,
-				version: 1,
-				code: 404,
-				error_code: 20,
-				message: 'BAD_COMMUNITY_DATA'
-			}
-		}).end({ pretty: true }));
-		return;
+		return respondCommunityError(response, 400, 20);
+	}
+
+	// Name must be at least 4 character long
+	if (request.body.name.length < 4) {
+		return respondCommunityError(response, 400, 20);
+	}
+
+	// Name must contain less than 5 numbers
+	const digitCount = (request.body.name.match(/\d/g) || []).length;
+	if (digitCount > 5) {
+		return respondCommunityError(response, 400, 20);
+	}
```

Comment:
This check is confusing to me. Why must the name contain less than 5 numbers? And is this supposed to be consecutive numbers, or 5 numbers _anywhere_ in the string? For example the string `a1b2c3d4e5f6` will count as "more than 5 numbers" with this regex

Reply:
I got this from here (MK8 tournament creation), i'm unsure how it was ruled back then, but i assumed it's 5 numbers total

![image](https://github.com/PretendoNetwork/miiverse-api/assets/39063367/5478e0e0-96c2-4d8a-9bca-8396c53018ce)

Reply:
Usually these things have checks on the client as well (for example when registering an account, the client checks the format of the username and password before sending it to the server). Does MK8 ever give you an error like that? Like does it try to hit the server if there's more than 6 numbers?

Reply:
I just checked, and the game doesn't prevent you from inputing more than five numbers, i also just saw this restriction should apply to the description as well